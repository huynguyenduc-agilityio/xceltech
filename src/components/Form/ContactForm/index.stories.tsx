import type { Meta, StoryObj } from '@storybook/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import ContactForm from '.';

const queryClient = new QueryClient();

const meta = {
  title: 'Components/ContactForm',
  component: ContactForm,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  render: () => (
    <QueryClientProvider client={queryClient}>
      <div className="flex items-center justify-center w-screen py-20 bg-blue-light">
        <div className="bg-white w-[988px] rounded-[15px]">
          <div className="p-10">
            <ContactForm />
          </div>
        </div>
      </div>
    </QueryClientProvider>
  ),
} satisfies Meta<typeof ContactForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
