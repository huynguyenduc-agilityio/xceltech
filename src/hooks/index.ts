export * from './useToast';
export * from './useConfirm';
export * from './useAuth';
export * from './useUser';
export * from './useLeave';
export * from './useAccount';
export * from './usePagination';
export * from './useNotification';
export * from './useJob';
export * from './useDocument';
export * from './useEducation';
export * from './useGuarantor';
export * from './useFamily';
export * from './useFinancial';
