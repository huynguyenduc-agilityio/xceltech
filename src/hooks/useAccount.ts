import { useQuery } from '@tanstack/react-query';

// Constants
import { accountsQueryKeys } from '@/constants';

// Types
import { IInfoUser } from '@/types';

// Services
import { getAccounts } from '@/services';

export const useGetLeaveAccounts = (userId: string) => {
  const { data: leaveAccounts, isFetching: isLeaveAccountsLoading } = useQuery({
    queryKey: [...accountsQueryKeys.list()] as const,
    queryFn: getAccounts,
    select: (data) =>
      data
        ?.filter(({ id }: { id: string }) => id !== userId)
        .map(({ id, firstName, lastName }: IInfoUser) => ({
          value: id,
          label: `${firstName} ${lastName}`,
        })),
  });

  return { leaveAccounts, isLeaveAccountsLoading };
};
