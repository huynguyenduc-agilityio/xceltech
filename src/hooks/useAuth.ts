import { useNavigate, useLocation } from 'react-router-dom';
import { useMutation } from '@tanstack/react-query';

//Types
import { IActiveUser } from '@/types';

// Constants
import {
  ADMIN_PAGE,
  AUTHENTICATION_PAGE,
  RoleAuthentication,
  USER_PAGE,
} from '@/constants';

// Services
import { activateAccount, loginUser, registerUser } from '@/services';

// Stores
import { useUserActions } from '@/stores';

export const useLogin = () => {
  const navigate = useNavigate();
  const location = useLocation();
  const { setUser } = useUserActions();

  const {
    mutateAsync: handleLoginUser,
    isPending: isLoginLoading,
    isError: isLoginError,
  } = useMutation({
    mutationFn: loginUser,
    onSuccess: (data) => {
      // Save data in local storage
      if (data) {
        const path = location.pathname;
        const role = data.user.role;
        const isAdmin =
          role === RoleAuthentication.Admin &&
          path === AUTHENTICATION_PAGE.ADMIN_SIGN_IN;

        const isUser =
          [RoleAuthentication.Employee, RoleAuthentication.Candidate].includes(
            role as RoleAuthentication,
          ) && path === AUTHENTICATION_PAGE.USER_SIGN_IN;

        if (isAdmin) {
          setUser(data);
          navigate(ADMIN_PAGE.DASHBOARD);
        }

        if (isUser) {
          setUser(data);
          navigate(USER_PAGE.DASHBOARD);
        }
      }
    },
  });

  return { isLoginLoading, isLoginError, handleLoginUser };
};

export const useRegister = () => {
  const { mutateAsync: handleRegisterUser, isPending: isRegisterLoading } =
    useMutation({
      mutationFn: registerUser,
    });

  return {
    handleRegisterUser,
    isRegisterLoading,
  };
};

export const useActivateAccount = ({ uidb64, token }: IActiveUser) => {
  const { mutateAsync: handleActivateUser, isPending: isActivateLoading } =
    useMutation({
      mutationFn: () => activateAccount({ uidb64, token }),
    });

  return {
    handleActivateUser,
    isActivateLoading,
  };
};
