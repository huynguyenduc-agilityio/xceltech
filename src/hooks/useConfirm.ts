import { useConfirmDialogStore } from '@/stores';

export const useConfirm = () => {
  return useConfirmDialogStore((state) => state.confirm);
};
