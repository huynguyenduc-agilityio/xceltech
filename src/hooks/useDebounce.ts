import { useCallback, useRef } from 'react';

export const useDebounce = <TParam = unknown>(
  callback: (...args: TParam[]) => void,
  delay = 600,
) => {
  const refTime = useRef<NodeJS.Timeout | null>(null);

  const debouncedFn = useCallback(
    (...args: TParam[]) => {
      if (refTime.current) clearTimeout(refTime.current);
      refTime.current = setTimeout(() => {
        callback(...args);
      }, delay);
    },
    [callback, delay],
  );

  return debouncedFn;
};
