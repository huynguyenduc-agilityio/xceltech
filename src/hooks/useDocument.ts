import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';

// Constants
import { documentsQueryKeys, MESSAGES } from '@/constants';

// Types
import { MutationType } from '@/types';

// Services
import {
  downloadAllDocuments,
  getDocuments,
  addDocument,
  editDocument,
} from '@/services';

// Utils
import { downloadBlobFile } from '@/utils';

export const useGetDocuments = () => {
  const { data: documents, isFetching: isDocumentsLoading } = useQuery({
    queryKey: documentsQueryKeys.list(),
    queryFn: getDocuments,
  });

  return { documents, isDocumentsLoading };
};

export const useDocumentMutation = ({ type }: { type: MutationType }) => {
  const queryClient = useQueryClient();

  const mutationFn = type === MutationType.Create ? addDocument : editDocument;

  const {
    mutateAsync: handleDocumentMutation,
    isPending: isDocumentMutationLoading,
  } = useMutation({
    mutationFn,
    onSuccess: () => {
      queryClient.invalidateQueries({
        queryKey: documentsQueryKeys.list(),
      });
    },
  });

  return { handleDocumentMutation, isDocumentMutationLoading };
};

export const useDownloadDocuments = () => {
  const { isPending: isLoading, mutateAsync: handleDownloadDocuments } =
    useMutation({
      mutationFn: downloadAllDocuments,
      onSuccess: (data) => {
        if (!data) {
          throw new Error(MESSAGES.COMMON.DOWNLOAD_FAILED);
        }

        downloadBlobFile(data, 'documents.zip');
      },
    });

  return { isLoading, handleDownloadDocuments };
};
