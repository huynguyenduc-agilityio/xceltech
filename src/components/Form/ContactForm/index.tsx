import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';

// Constants
import { MESSAGES } from '@/constants';

// Types
import { ToastStatus, IInfoUser, ContactFormValues } from '@/types';

// Components
import {
  Button,
  Form,
  FormControl,
  FormField,
  FormItem,
  FormMessage,
  Label,
  Textarea,
  TextField,
} from '@/components';

// Hooks
import { useToast, useUpdateInfoUser } from '@/hooks';

// Utils
import { contactSchema } from '@/utils/schemas/employee';

export interface IContactForm {
  initialValues?: Partial<IInfoUser>;
}

const ContactForm = ({ initialValues }: IContactForm) => {
  const { toast } = useToast();
  const { handleUpdateInfoUser, isUpdateInfoLoading } = useUpdateInfoUser();

  const {
    phone = '',
    email = '',
    contact = { phoneNum2: '', cityOfResidence: '', residentialAddress: '' },
  } = initialValues || {};

  const defaultValues = {
    phone,
    email,
    phoneNum2: contact?.phoneNum2 || '',
    cityOfResidence: contact?.cityOfResidence || '',
    residentialAddress: contact?.residentialAddress || '',
  };

  const form = useForm<ContactFormValues>({
    mode: 'onBlur',
    reValidateMode: 'onChange',
    resolver: zodResolver(contactSchema),
    defaultValues,
  });

  const {
    control,
    handleSubmit,
    reset,
    getValues,
    formState: { isValid, isDirty, isSubmitting },
  } = form;

  const onSubmit = async (data: ContactFormValues) => {
    const payload: Partial<IInfoUser> = {
      phone: data.phone,
      email: data.email,
      contact: {
        phoneNum2: data.phoneNum2,
        cityOfResidence: data.cityOfResidence,
        residentialAddress: data.residentialAddress,
      },
    };

    try {
      await handleUpdateInfoUser(payload);

      toast({
        status: ToastStatus.Success,
        title: MESSAGES.COMMON.UPDATE_SUCCESS('Contact'),
      });
    } catch {
      toast({
        status: ToastStatus.Error,
        title: MESSAGES.COMMON.UPDATE_FAILED('Contact'),
      });
    } finally {
      reset(getValues());
    }
  };

  const disableSubmit = !isDirty || !isValid || isSubmitting;

  return (
    <Form {...form}>
      <form
        onSubmit={handleSubmit(onSubmit)}
        className="w-full px-10 py-16 text-black-default space-y-10"
      >
        <div className="flex items-start gap-14">
          <FormField
            control={control}
            name="phone"
            render={({ field, fieldState: { error } }) => (
              <TextField
                label="Phone Number 1"
                variant="primary"
                className="space-y-5"
                labelClassName="text-md"
                inputClassName="h-[68px] px-12 rounded-[15px]"
                placeholder="Enter your phone number 1"
                {...field}
                errorMessage={error?.message}
              />
            )}
          />
          <FormField
            control={control}
            name="phoneNum2"
            render={({ field, fieldState: { error } }) => (
              <TextField
                label="Phone Number 2"
                variant="primary"
                className="space-y-5"
                labelClassName="text-md"
                inputClassName="h-[68px] px-12 rounded-[15px]"
                placeholder="Enter your phone number 2"
                {...field}
                errorMessage={error?.message}
              />
            )}
          />
        </div>
        <FormField
          control={control}
          name="email"
          render={({ field, fieldState: { error } }) => (
            <TextField
              label="E-mail Address"
              variant="primary"
              className="space-y-5"
              labelClassName="text-md"
              inputClassName="h-[68px] px-12 rounded-[15px]"
              placeholder="Enter your email address"
              {...field}
              errorMessage={error?.message}
            />
          )}
        />
        <FormField
          control={control}
          name="cityOfResidence"
          render={({ field, fieldState: { error } }) => (
            <TextField
              label="City of residence"
              variant="primary"
              className="space-y-5"
              labelClassName="text-md"
              inputClassName="h-[68px] max-w-[431px] px-12 rounded-[15px]"
              placeholder="Enter your city of residence"
              {...field}
              errorMessage={error?.message}
            />
          )}
        />

        <FormField
          control={control}
          name="residentialAddress"
          render={({ field }) => (
            <FormItem>
              <Label className="text-md">Residential Address</Label>
              <FormControl>
                <Textarea
                  {...field}
                  className="h-[122px] px-9 rounded-[15px]"
                  placeholder="Enter your address"
                />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />
        <div className="w-full h-[70px]">
          <Button
            type="submit"
            className="w-[364px] bg-green-primary"
            disabled={disableSubmit}
            isLoading={isUpdateInfoLoading}
          >
            Update
          </Button>
        </div>
      </form>
    </Form>
  );
};

export default ContactForm;
