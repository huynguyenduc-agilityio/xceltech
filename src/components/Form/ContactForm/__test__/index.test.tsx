import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import ContactForm from '..';

// Hooks
import { useToast, useUpdateInfoUser } from '@/hooks';

const mockHandleUpdateInfoUser = jest.fn();
const mockToast = jest.fn();
const mockHandleUpdateInfoUserError = jest
  .fn()
  .mockRejectedValue(new Error('Network error'));

jest.mock('@/hooks', () => ({
  useToast: jest.fn(),
  useUpdateInfoUser: jest.fn(),
}));

const renderComponent = () => {
  const queryClient = new QueryClient();

  return render(
    <QueryClientProvider client={queryClient}>
      <ContactForm />
    </QueryClientProvider>,
  );
};

describe('ContactForm Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    (useToast as jest.Mock).mockReturnValue({ toast: mockToast });
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isFamilyMutationLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUser,
    });
  });

  it('should match snapshot', () => {
    const container = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('handle input value change and form submit', async () => {
    const { getByPlaceholderText, getByRole } = renderComponent();
    const phone = getByPlaceholderText('Enter your phone number 1');
    const phoneNum2 = getByPlaceholderText('Enter your phone number 2');
    const emailInput = getByPlaceholderText('Enter your email address');
    const cityInput = getByPlaceholderText('Enter your city of residence');
    const addressInput = getByPlaceholderText('Enter your address');
    const submitButton = getByRole('button', { name: 'Update' });

    fireEvent.change(phone, { target: { value: '1234567890' } });
    fireEvent.change(phoneNum2, { target: { value: '9876543210' } });
    fireEvent.change(emailInput, { target: { value: '2t6d0@example.com' } });
    fireEvent.change(cityInput, { target: { value: 'New York' } });
    fireEvent.change(addressInput, { target: { value: '123 Main St' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockHandleUpdateInfoUser).toHaveBeenCalled();
    });
  });

  it('handles submit error gracefully', async () => {
    (useUpdateInfoUser as jest.Mock).mockReturnValue({
      isLoading: false,
      handleUpdateInfoUser: mockHandleUpdateInfoUserError,
    });

    const { getByPlaceholderText, getByRole } = renderComponent();
    const phone = getByPlaceholderText('Enter your phone number 1');
    const phoneNum2 = getByPlaceholderText('Enter your phone number 2');
    const emailInput = getByPlaceholderText('Enter your email address');
    const cityInput = getByPlaceholderText('Enter your city of residence');
    const addressInput = getByPlaceholderText('Enter your address');
    const submitButton = getByRole('button', { name: 'Update' });

    fireEvent.change(phone, { target: { value: '1234567890' } });
    fireEvent.change(phoneNum2, { target: { value: '9876543210' } });
    fireEvent.change(emailInput, { target: { value: '2t6d0@example.com' } });
    fireEvent.change(cityInput, { target: { value: 'New York' } });
    fireEvent.change(addressInput, { target: { value: '123 Main St' } });

    await userEvent.click(submitButton);

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalled();
    });
  });

  it('should show error message fields', async () => {
    const { getByPlaceholderText, getByText } = renderComponent();
    const phone = getByPlaceholderText('Enter your phone number 1');
    const phoneNum2 = getByPlaceholderText('Enter your phone number 2');
    const emailInput = getByPlaceholderText('Enter your email address');
    const cityInput = getByPlaceholderText('Enter your city of residence');
    const addressInput = getByPlaceholderText('Enter your address');

    fireEvent.change(phone, { target: { value: '' } });
    fireEvent.change(phoneNum2, { target: { value: '' } });
    fireEvent.change(emailInput, { target: { value: '' } });
    fireEvent.change(cityInput, { target: { value: '' } });
    fireEvent.change(addressInput, { target: { value: '' } });

    fireEvent.blur(phone);
    fireEvent.blur(phoneNum2);
    fireEvent.blur(emailInput);
    fireEvent.blur(cityInput);
    fireEvent.blur(addressInput);

    await waitFor(() => {
      expect(getByText('Phone Number 1 is required!')).toBeInTheDocument();
      expect(getByText('Email is required!')).toBeInTheDocument();
      expect(getByText('City of residence is required!')).toBeInTheDocument();
      expect(getByText('Residential Address is required!')).toBeInTheDocument();
    });
  });
});
