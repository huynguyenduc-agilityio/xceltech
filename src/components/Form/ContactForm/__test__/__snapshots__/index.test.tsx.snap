// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="w-full px-10 py-16 text-black-default space-y-10"
      >
        <div
          class="flex items-start gap-14"
        >
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Phone Number 1-label"
              >
                Phone Number 1
              </label>
              <input
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Phone Number 1-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r0:-form-item"
                name="phone"
                placeholder="Enter your phone number 1"
                value=""
              />
            </div>
          </div>
          <div
            class="grid w-full gap-2"
          >
            <div
              class="w-full space-y-5"
            >
              <label
                class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
                id="Phone Number 2-label"
              >
                Phone Number 2
              </label>
              <input
                aria-describedby=":r1:-form-item-description"
                aria-invalid="false"
                aria-labelledby="Phone Number 2-label"
                autocomplete="off"
                class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
                data-testid="input-element"
                id=":r1:-form-item"
                name="phoneNum2"
                placeholder="Enter your phone number 2"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="grid w-full gap-2"
        >
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              id="E-mail Address-label"
            >
              E-mail Address
            </label>
            <input
              aria-describedby=":r2:-form-item-description"
              aria-invalid="false"
              aria-labelledby="E-mail Address-label"
              autocomplete="off"
              class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-testid="input-element"
              id=":r2:-form-item"
              name="email"
              placeholder="Enter your email address"
              value=""
            />
          </div>
        </div>
        <div
          class="grid w-full gap-2"
        >
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              id="City of residence-label"
            >
              City of residence
            </label>
            <input
              aria-describedby=":r3:-form-item-description"
              aria-invalid="false"
              aria-labelledby="City of residence-label"
              autocomplete="off"
              class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] max-w-[431px] px-12 rounded-[15px]"
              data-testid="input-element"
              id=":r3:-form-item"
              name="cityOfResidence"
              placeholder="Enter your city of residence"
              value=""
            />
          </div>
        </div>
        <div
          class="space-y-2 w-full"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
          >
            Residential Address
          </label>
          <textarea
            aria-describedby=":r4:-form-item-description"
            aria-invalid="false"
            class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
            id=":r4:-form-item"
            name="residentialAddress"
            placeholder="Enter your address"
          />
        </div>
        <div
          class="w-full h-[70px]"
        >
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
            disabled=""
            type="submit"
          >
            Update
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="w-full px-10 py-16 text-black-default space-y-10"
    >
      <div
        class="flex items-start gap-14"
      >
        <div
          class="grid w-full gap-2"
        >
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              id="Phone Number 1-label"
            >
              Phone Number 1
            </label>
            <input
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              aria-labelledby="Phone Number 1-label"
              autocomplete="off"
              class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-testid="input-element"
              id=":r0:-form-item"
              name="phone"
              placeholder="Enter your phone number 1"
              value=""
            />
          </div>
        </div>
        <div
          class="grid w-full gap-2"
        >
          <div
            class="w-full space-y-5"
          >
            <label
              class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
              id="Phone Number 2-label"
            >
              Phone Number 2
            </label>
            <input
              aria-describedby=":r1:-form-item-description"
              aria-invalid="false"
              aria-labelledby="Phone Number 2-label"
              autocomplete="off"
              class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
              data-testid="input-element"
              id=":r1:-form-item"
              name="phoneNum2"
              placeholder="Enter your phone number 2"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="grid w-full gap-2"
      >
        <div
          class="w-full space-y-5"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            id="E-mail Address-label"
          >
            E-mail Address
          </label>
          <input
            aria-describedby=":r2:-form-item-description"
            aria-invalid="false"
            aria-labelledby="E-mail Address-label"
            autocomplete="off"
            class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] px-12 rounded-[15px]"
            data-testid="input-element"
            id=":r2:-form-item"
            name="email"
            placeholder="Enter your email address"
            value=""
          />
        </div>
      </div>
      <div
        class="grid w-full gap-2"
      >
        <div
          class="w-full space-y-5"
        >
          <label
            class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
            id="City of residence-label"
          >
            City of residence
          </label>
          <input
            aria-describedby=":r3:-form-item-description"
            aria-invalid="false"
            aria-labelledby="City of residence-label"
            autocomplete="off"
            class="font-productsans flex w-full py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none disabled:cursor-not-allowed disabled:opacity-50 bg-blue-light h-[68px] max-w-[431px] px-12 rounded-[15px]"
            data-testid="input-element"
            id=":r3:-form-item"
            name="cityOfResidence"
            placeholder="Enter your city of residence"
            value=""
          />
        </div>
      </div>
      <div
        class="space-y-2 w-full"
      >
        <label
          class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-md"
        >
          Residential Address
        </label>
        <textarea
          aria-describedby=":r4:-form-item-description"
          aria-invalid="false"
          class="flex min-h-[80px] w-full rounded-regular bg-blue-light py-4 text-lg text-black-smoky placeholder:text-gray-soft outline-none resize-none disabled:cursor-not-allowed disabled:opacity-50 h-[122px] px-9 rounded-[15px]"
          id=":r4:-form-item"
          name="residentialAddress"
          placeholder="Enter your address"
        />
      </div>
      <div
        class="w-full h-[70px]"
      >
        <button
          class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white h-[70px] py-6 w-[364px] bg-green-primary"
          disabled=""
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
